src/Funogram/Tools.fs

```diff
@@ -54,6 +55,31 @@ let internal parseJson<'a> (data: byte[]) =
     let message = sprintf "%s in %s" ex.Message json
     ArgumentException(message, ex) |> raise
 
+let internal parseJsonStream<'a> (data: Stream) =
+  try
+    JsonSerializer.Deserialize<'a>(data, resolver) |> Ok
+  with ex ->
+    if data.CanSeek then 
+      data.Seek(0L, SeekOrigin.Begin) |> ignore
+      use sr = new StreamReader(data)
+      let message = sprintf "%s in %s" ex.Message (sr.ReadToEnd())
+      ArgumentException(message, ex) :> Exception |> Result.Error
+    else
+      Exception("Can't parse json") |> Result.Error
+
+let internal parseJsonStreamApiResponse<'a> (data: Stream) =
+  match parseJsonStream<Types.ApiResponse<'a>> data with
+  | Ok x when x.Ok && x.Result.IsSome -> Ok x.Result.Value
+
+  | Ok x when x.Description.IsSome && x.ErrorCode.IsSome -> 
+    Error { Description = x.Description.Value; ErrorCode = x.ErrorCode.Value }
+
+  | Error e -> 
+    Error { Description = e.Message; ErrorCode = -1 }
+
+  | _ -> 
+    Error { Description = "Unknown error"; ErrorCode = -1 }
```

Comment:
Would it be a good idea to at least try to stringify this unknown response here? As I understand, this case could match if there's `Ok x`, but neither `x.Result` nor `x.Description` is set. Which would be odd, yes, but still possible according to the types, and probably even having somewhat useful string representation.

E.g. `| other -> Error { Description = sprintf "Unknown error: %s" other; ErrorCode = -1 }`